src/main/java/de/mediathekview/mserver/crawler/phoenix/tasks/PhoenixFilmDetailTask.java

```diff
@@ -0,0 +1,148 @@
+package de.mediathekview.mserver.crawler.phoenix.tasks;
+
+import com.google.gson.reflect.TypeToken;
+import de.mediathekview.mlib.daten.Film;
+import de.mediathekview.mlib.daten.Sender;
+import de.mediathekview.mserver.crawler.basic.AbstractCrawler;
+import de.mediathekview.mserver.crawler.basic.AbstractRecrusivConverterTask;
+import de.mediathekview.mserver.crawler.basic.CrawlerUrlDTO;
+import de.mediathekview.mserver.crawler.phoenix.PhoenixConstants;
+import de.mediathekview.mserver.crawler.phoenix.parser.PhoenixFilmDetailDeserializer;
+import de.mediathekview.mserver.crawler.phoenix.parser.PhoenixFilmDetailDto;
+import de.mediathekview.mserver.crawler.phoenix.parser.PhoenixFilmXmlHandler;
+import de.mediathekview.mserver.crawler.zdf.DownloadDtoFilmConverter;
+import de.mediathekview.mserver.crawler.zdf.ZdfConstants;
+import de.mediathekview.mserver.crawler.zdf.ZdfVideoUrlOptimizer;
+import de.mediathekview.mserver.crawler.zdf.json.DownloadDto;
+import de.mediathekview.mserver.crawler.zdf.json.ZdfDownloadDtoDeserializer;
+import de.mediathekview.mserver.crawler.zdf.tasks.ZdfTaskBase;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import javax.ws.rs.client.WebTarget;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.xml.sax.SAXException;
+
+public class PhoenixFilmDetailTask extends ZdfTaskBase<Film, CrawlerUrlDTO> {
+
+  private static final Logger LOG = LogManager.getLogger(PhoenixFilmDetailTask.class);
+
+  private static final Type OPTIONAL_FILM_DETAIL_DTO_TYPE_TOKEN = new TypeToken<Optional<PhoenixFilmDetailDto>>() {
+  }.getType();
+  private static final Type OPTIONAL_DOWNLOAD_DTO_TYPE_TOKEN = new TypeToken<Optional<DownloadDto>>() {
+  }.getType();
+
+  private final String filmDetailHost;
+  private final String videoDetailHost;
+
+  private final ZdfVideoUrlOptimizer optimizer = new ZdfVideoUrlOptimizer();
+
+  public PhoenixFilmDetailTask(AbstractCrawler aCrawler,
+      ConcurrentLinkedQueue<CrawlerUrlDTO> aUrlToCrawlDTOs, Optional<String> aAuthKey,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDTOs" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)